keps/sig-node/compute-device-assignment.md

```diff
@@ -57,15 +59,33 @@ In this document we will discuss the motivation and code changes required for in
 
 ![device monitoring architecture](https://user-images.githubusercontent.com/3262098/43926483-44331496-9bdf-11e8-82a0-14b47583b103.png)
 
+### Device aware CNI plugin
+As soon as this interface has been introduced it was used by CNI plugins like [kuryr-kubernetes](https://review.opendev.org/#/c/651580/) in couple with [intel-sriov-device-plugin](https://github.com/intel/sriov-network-device-plugin) to correctly define which devices were assigned to the pod.
 
-## Changes
+### Topology aware scheduling
+This interface can be used to collect allocated resources with information about the NUMA topology of the worker node.
+This interface can be used to the available resources on the worker node. The kubelet is the best source of information because it manages the device plugins. The information can then be used in NUMA aware scheduling.
 
-Add a v1alpha1 Kubelet GRPC service, at `/var/lib/kubelet/pod-resources/kubelet.sock`, which returns information about the kubelet's assignment of devices to containers. It obtains this information from the internal state of the kubelet's Device Manager. The GRPC Service returns a single PodResourcesResponse, which is shown in proto below:
+## Changes
+Add a v1alpha1 Kubelet GRPC service, at `/var/lib/kubelet/pod-resources/kubelet.sock`, which returns information about
+- the devices which kubelet knows about from the device plugins.
+- the kubelet's assignment of devices and cpus with NUMA id to containers.
+The GRPC service obtains this information from the internal state of the kubelet's Device Manager and CPU Manager respectively.
+The GRPC Service is described in proto below:
 ```protobuf
 // PodResources is a service provided by the kubelet that provides information about the
 // node resources consumed by pods and containers on the node
 service PodResources {
     rpc List(ListPodResourcesRequest) returns (ListPodResourcesResponse) {}
+    rpc GetAvailableResources(AvailableResourcesRequest) returns (AvailableResourcesResponse) {}
+}
+
+message AvailableResourcesRequest {}
+
+// AvailableResourcesResponses contains informations about all the devices known by the kubelet
+message AvailableResourcesResponse {
+    repeated ContainerDevices devices = 1;
+    repeated int64 cpu_ids = 2;
```

Comment:
how would this evolve with the memory manager kep?

what resources would you propose to expose related to memory as available?
https://github.com/kubernetes/enhancements/pull/1203

Reply:
ContainerDevices struct contains resource_name -> v1.ResourceMemory could be there. But, I think, we need to mention it in KEP.

Reply:
I think the only issue here is the ordering between this KEP and the memory manager KEP. We can mention the plans to extend when new `HintProviders` are added.

Reply:
My opinion on supporting memory resources - interface proposed here could represent memory resources w/o any further modification of the interface.